include/exadg/structure/time_integration/time_int_gen_alpha.cpp

```diff
@@ -146,12 +146,12 @@ TimeIntGenAlpha<dim, Number>::do_timestep_solve()
   else
     displacement_np = displacement_last_iter;
 
+  bool const update_preconditioner =
+    this->param.update_preconditioner &&
+    ((this->time_step_number - 1) % this->param.update_preconditioner_every_time_steps == 0);
+
```

Comment:
preconditioner might be updated for linear (instat) solve as well -> scaling for adaptive timestepping and large dt variations?